contracts/gnosis_safe_modules/DelayedRecoveryModule.sol

```diff
@@ -0,0 +1,627 @@
+
+/**
+ * @title Allows to replace an owner without Safe confirmations
+ *        if the recovery owner and the recovery controller approves
+ *        replacement.
+ *
+ * @dev The contract is a module for gnosis safe multisig contract.
+ *      Gnosis safe multisig's modules assume a common interface which
+ *      in the current gnosis safe implementation is inside Module.sol
+ *      contract of Gnosis Safe. Instead of inheriting from this contract
+ *      (not to include gnosis contracts into build process)
+ *      GnosisSafeModule.sol and GnosisSafeModuleManagerInterface.sol contracts
+ *      are introduced that contains the required public interfaces.
+ */
+contract DelayedRecoveryModule is GnosisSafeModule {
+
+    /* Events */
+
+    event RecoveryInitiated(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    );
+
+    event RecoveryExecuted(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    );
+
+    event RecoveryAborted(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    );
+
+    event ResetRecoveryOwner(
+        address _oldRecoveryOwner,
+        address _newRecoveryOwner
+    );
+
+
+    /* Constants */
+
+    string public constant NAME = "Delayed Recovery Module";
+
+    string public constant VERSION = "0.1.0";
+
+    bytes32 public constant DOMAIN_SEPARATOR_TYPEHASH = keccak256(
+        "EIP712Domain(address delayedRecoveryModule)"
+    );
+
+    bytes32 public constant INITIATE_RECOVERY_STRUCT_TYPEHASH = keccak256(
+        "InitiateRecoveryStruct(address prevOwner,address oldOwner,address newOwner)"
+    );
+
+    bytes32 public constant EXECUTE_RECOVERY_STRUCT_TYPEHASH = keccak256(
+        "ExecuteRecoveryStruct(address prevOwner,address oldOwner,address newOwner)"
+    );
+
+    bytes32 public constant ABORT_RECOVERY_STRUCT_TYPEHASH = keccak256(
+        "AbortRecoveryStruct(address prevOwner,address oldOwner,address newOwner)"
+    );
+
+    bytes32 public constant RESET_RECOVERY_OWNER_STRUCT_TYPEHASH = keccak256(
+        "ResetRecoveryOwnerStruct(address oldRecoveryOwner,address newRecoveryOwner)"
+    );
+
+
+    /* Structs */
+
+    struct RecoveryInfo {
+        address prevOwner;
+        address oldOwner;
+        address newOwner;
+        uint256 initiationBlockHeight;
+        bool initiated;
+    }
+
+
+    /* Storage */
+
+    bytes32 public domainSeparator;
+
+    address public recoveryController;
+
+    address public recoveryOwner;
+
+    uint256 public recoveryBlockDelay;
+
+    RecoveryInfo public activeRecoveryInfo;
+
+
+    /* Modifiers */
+
+    modifier onlyRecoveryController()
+    {
+        require(
+            msg.sender == recoveryController,
+            "Only recovery controller is allowed to call."
+        );
+
+        _;
+    }
+
+    modifier activeRecovery()
+    {
+        require(
+            activeRecoveryInfo.initiated,
```

Comment:
we can check `require(activeRecoveryInfo.initiationBlockHeight > 0)`

Reply:
better to use `executionBlockHeight` to ensure it is never zero

Reply:
Done.